api/v1beta1/openstackbaremetalset.go

```diff
@@ -131,8 +131,13 @@ func VerifyBaremetalSetScaleUp(
 				mismatch = true
 			}
 
-			if baremetalHost.Spec.Online {
-				l.Info("BaremetalHost cannot be used because it is already online", "BMH", baremetalHost.ObjectMeta.Name)
+			if baremetalHost.Spec.ExternallyProvisioned {
```

Comment:
As `consumerRef` is optional,  should we also check if `image.URL` has been set?

Reply:
I guess, when looking I wonder - could we use the [WasProvisioned()](https://github.com/metal3-io/baremetal-operator/blob/b6aa2579347fe05b2a7e3f8728f7bbc0498b664d/apis/metal3.io/v1alpha1/baremetalhost_types.go#L1036C28-L1036C42) ?
```
if baremetalHost.WasProvisioned() {
	l.Info("BaremetalHost cannot be used because it was already provisioned", "BMH", baremetalHost.ObjectMeta.Name)
	mismatch = true
}
```